src/System.Management.Automation/engine/regex.cs

```diff
@@ -148,7 +146,26 @@ private void Init()
             {
                 if (Pattern.Length == 1 && Pattern[0] == '*')
                 {
-                    _isMatch = _ => true;
+                    _isMatch = s_matchAll;
+                }
+                else if (Pattern.IndexOfAny(s_specialChars) == -1)
+                {
+                    // No special characters present in the pattern, so we can just do a string comparison.
+                    StringComparison stringComparison;
+                    if (Options.HasFlag(WildcardOptions.IgnoreCase))
```

Comment:
We could unroll the if-s to switch for better readability.

Reply:
The `WildcardOptions` enum has `[Flags]` declared, so `Options` could be the combination of multiple `WildcardOptions` members. That makes it not friendly to `switch`. For example, the `Compile` member is completely ignored in this scenario, though it could be declared in `Options`.